src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -178,35 +206,48 @@ public Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectResto
         /// <param name="token"></param>
         /// <remarks>Exactly one of projectRestoreInfos has to null.</remarks>
         /// <returns>The task that scheduled restore</returns>
-        private async Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectRestoreInfo projectRestoreInfo, IVsProjectRestoreInfo2 projectRestoreInfo2, CancellationToken token)
+        private async Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectRestoreInfo3 projectRestoreInfo, CancellationToken token)
         {
             if (string.IsNullOrEmpty(projectUniqueName))
             {
                 throw new ArgumentException(Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(projectUniqueName));
             }
 
-            if (projectRestoreInfo == null && projectRestoreInfo2 == null)
+            if (projectRestoreInfo == null)
             {
                 throw new ArgumentNullException(nameof(projectRestoreInfo));
             }
 
-            if (projectRestoreInfo != null && projectRestoreInfo2 != null)
+            if (string.IsNullOrEmpty(projectRestoreInfo.BaseIntermediatePath))
```

Comment:
Guessing this is supposed to be an improvement?

Reply:
It's "insurance" against a project system with a bad implementation.

If the value is null, it'll be passed into `Path.Combine` and throw an `ArgumentNullException`, but the exception message will be non-obvious that it's a problem from the caller violating NuGet's API contract. By checking and providing a clear error message, it'll be easier for 3rd party project systems to see that they're not meeting our API's expectations.

If the value is an empty string, we're going to write the assets file, and all NuGet's other output files, to the project directory, which is also clearly wrong.